src/modules/launcher/Plugins/Microsoft.Plugin.WindowWalker/Components/ResultHelper.cs

```diff
@@ -34,7 +34,12 @@ internal static List<Result> GetResultList(List<SearchResult> searchControllerRe
                 resultsList.Add(new Result()
                 {
                     Title = x.Result.Title,
-                    IcoPath = icon,
+                    IcoPath = WindowWalkerSettings.Instance.UseWindowIconInResults ? string.Empty : icon,
+                    Icon = WindowWalkerSettings.Instance.UseWindowIconInResults ? () =>
+                    {
+                        return x.Result.WindowIcon ?? x.Result.Process.ProcessIcon;
```

Comment:
Why do we need two caches / icon sources? A window should have only one icon?

(I am worry about slow down PT Run search.)

Reply:
So we have one icon from window and a (fallback) icon for the process. Then the two caches make sense.